onebusaway-android/src/main/java/org/onebusaway/android/util/UIUtils.java

```diff
@@ -1447,13 +1448,24 @@ public static void closeKeyboard(Context context, View v) {
             allIds.add(s.getId());
         }
 
+        // Do the same for filtered routes
+        HashSet<String> filterIds = new HashSet<>();
+        if (filter != null) {
```

Comment:
I would make this `if (filter != null && !filter.isEmpty()) {` in case someone passes in an empty `filter` list by accident.